receiver/httpreceiver/logs.go

```diff
@@ -0,0 +1,232 @@
+		serverSettings:    cfg.ServerSettings,
+		telemetrySettings: params.TelemetrySettings,
+		consumer:          consumer,
+		wg:                &sync.WaitGroup{},
+		logger:            params.Logger,
+	}, nil
+}
+
+// Start calls startListening
+func (r *httpLogsReceiver) Start(ctx context.Context, host component.Host) error {
+	return r.startListening(ctx, host)
+}
+
+// startListening starts serve on the server using TLS depending on receiver configuration
+func (r *httpLogsReceiver) startListening(_ context.Context, host component.Host) error {
+	r.logger.Debug("starting receiver HTTP server")
+	var err error
+	r.server, err = r.serverSettings.ToServer(host, r.telemetrySettings, r)
+	if err != nil {
+		return err
+	}
+
+	listener, err := r.serverSettings.ToListener()
+	if err != nil {
+		return err
+	}
+
+	r.wg.Add(1)
+	go func() {
+		defer r.wg.Done()
+
+		if r.serverSettings.TLSSetting != nil {
+			r.logger.Debug("starting ServeTLS",
+				zap.String("address", r.serverSettings.Endpoint),
+				zap.String("cert_file", r.serverSettings.TLSSetting.CertFile),
+				zap.String("key_file", r.serverSettings.TLSSetting.KeyFile),
+			)
+
+			err := r.server.ServeTLS(listener, r.serverSettings.TLSSetting.CertFile, r.serverSettings.TLSSetting.KeyFile)
+			r.logger.Debug("ServeTLS done")
+			if err != http.ErrServerClosed {
+				r.logger.Error("ServeTLS failed", zap.Error(err))
+				host.ReportFatalError(err)
+			}
+		} else {
+			r.logger.Debug("starting to serve",
+				zap.String("address", r.serverSettings.Endpoint),
+			)
+
+			err := r.server.Serve(listener)
+			r.logger.Debug("Serve done")
+			if err != http.ErrServerClosed {
+				r.logger.Error("Serve failed", zap.Error(err))
+				host.ReportFatalError(err)
+			}
+		}
+	}()
+
+	return nil
+}
+
+// Shutdown calls shutdownListener
+func (r *httpLogsReceiver) Shutdown(ctx context.Context) error {
+	return r.shutdownListener(ctx)
+}
+
+// shutdownLIstener tells the server to stop serving and waits for it to stop
+func (r *httpLogsReceiver) shutdownListener(ctx context.Context) error {
+	r.logger.Debug("shutting down server")
+
+	if err := r.server.Shutdown(ctx); err != nil {
+		return err
+	}
+	r.logger.Debug("waiting for shutdown to complete")
+	r.wg.Wait()
+	return nil
+
+}
+
+// handleRequest is the function the server uses for requests; calls ConsumeLogs
+func (r *httpLogsReceiver) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	// path was configured && this req.URL does not match it
+	if r.path != "" && req.URL.Path != r.path {
+		rw.WriteHeader(http.StatusNotFound)
+		r.logger.Debug("received request to path that does not match the configured path", zap.String("request path", req.URL.Path))
+		return
+	}
+
+	// read in request body
+	r.logger.Debug("reading in request body")
+	payload, err := io.ReadAll(req.Body)
+	if err != nil {
+		rw.WriteHeader(http.StatusUnprocessableEntity)
+		r.logger.Debug("failed to read logs payload", zap.Error(err), zap.String("remote", req.RemoteAddr))
```

Comment:
This should probably be an Error log so customers can see requests are not being processed.